test/cctest/test_node_postmortem_metadata.cc

```diff
@@ -0,0 +1,135 @@
+#include "node_postmortem_metadata.cc"
+
+#include "gtest/gtest.h"
+#include "node.h"
+#include "node_internals.h"
+#include "node_test_fixture.h"
+#include "req_wrap-inl.h"
+#include "tracing/agent.h"
+#include "v8.h"
+
+const int node::Environment::kContextEmbedderDataIndex;
+
+
+class DebugSymbolsTest : public EnvironmentTestFixture {};
+
+
+class TestHandleWrap : public node::HandleWrap {
+ public:
+  size_t self_size() const override { return sizeof(*this); }
+
+  TestHandleWrap(node::Environment* env,
+                 v8::Local<v8::Object> object,
+                 uv_handle_t* handle)
+      : node::HandleWrap(env,
+                         object,
+                         handle,
+                         node::AsyncWrap::PROVIDER_TIMERWRAP) {}
+};
+
+
+class TestReqWrap : public node::ReqWrap<uv_req_t> {
+ public:
+  size_t self_size() const override { return sizeof(*this); }
+
+  TestReqWrap(node::Environment* env, v8::Local<v8::Object> object)
+      : node::ReqWrap<uv_req_t>(env,
+                                object,
+                                node::AsyncWrap::PROVIDER_TIMERWRAP) {}
+};
+
+TEST_F(DebugSymbolsTest, ContextEmbedderDataIndex) {
+  EXPECT_EQ(nodedbg_const_Environment__kContextEmbedderDataIndex,
+            node::Environment::kContextEmbedderDataIndex);
+}
+
+TEST_F(DebugSymbolsTest, BaseObjectPersistentHandle) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env{handle_scope, argv, this};
+
+  v8::Local<v8::Object> object = v8::Object::New(isolate_);
+  node::BaseObject obj(*env, object);
+
+  auto expected = reinterpret_cast<uintptr_t>(&obj.persistent());
+  auto calculated = reinterpret_cast<uintptr_t>(&obj) +
+      nodedbg_offset_BaseObject__persistent_handle_;
+  EXPECT_EQ(expected, calculated);
+
+  obj.persistent().Reset();  // ~BaseObject() expects an empty handle.
+}
+
+
+TEST_F(DebugSymbolsTest, EnvironmentHandleWrapQueue) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env{handle_scope, argv, this};
+
+  auto expected = reinterpret_cast<uintptr_t>((*env)->handle_wrap_queue());
+  auto calculated = reinterpret_cast<uintptr_t>(*env) +
+      nodedbg_offset_Environment__handle_wrap_queue_;
+  EXPECT_EQ(expected, calculated);
+}
+
+TEST_F(DebugSymbolsTest, EnvironmentReqWrapQueue) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env{handle_scope, argv, this};
+
+  auto expected = reinterpret_cast<uintptr_t>((*env)->req_wrap_queue());
+  auto calculated = reinterpret_cast<uintptr_t>(*env) +
+      nodedbg_offset_Environment__req_wrap_queue_;
+  EXPECT_EQ(expected, calculated);
+}
+
+TEST_F(DebugSymbolsTest, HandleWrapList) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env{handle_scope, argv, this};
+
+  uv_handle_t handle;
```

Comment:
Can you use an initialized handle, like a `uv_tcp_t`?  Passing an uninitialized handle to HandleWrap may work now but that's not something I want to debug in the future.